### PR TITLE
Minor search docs tidy

### PR DESCRIPTION
**Pull request by hflynn**
See https://trello.com/c/Hq7F0yrm/190-rewrite-developer-documentation-for-search
Quick PR to try and clean up main nitpicks, if anyone has further suggestions for improvements or spots other out-of-date content, please comment and I'll do a wider edit.

Staged at https://www.openmicroscopy.org/site/support/omero5.3-staging/developers/Modules/Search.html

### CODE REVIEW

File: omero/developers/Modules/searchfieldnames.tsv [hunk 1]
```diff
@@ -11,7 +11,7 @@ details.updateEvent.id	Id of the Event of this objects last modification
 details.updateEvent.time	When that Event took place
 details.permissions	Permissions in the form `rwrwrw` or `rw-`
 tag	Contents from a ``TagAnnotation``.
-annotation	Contents from any annotations, including ``TagAnnotation`` and any ``TextAnnotation`` on another ``TextAnnotation`` (a.k.a. a `description`). Non-string annotations like file annotations are not covered by this definition and are handled separately. See below.
+annotation	Contents from annotations, including ``TagAnnotation`` and any ``TextAnnotation`` on another ``TextAnnotation`` (a.k.a. a `description`). Non-string annotations like ``FileAnnotations`` are not covered by this definition and are handled separately. See below.
```
Comment (mtbc) [inline omero/developers/Modules/searchfieldnames.tsv]: I am not sure we should have a _literal_ plural `FileAnnotations` as the only literal form in the model/code is the singular.

File: omero/developers/Modules/Search.txt [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
 
-Beginning with 3.0-Beta3, the OMERO server will use
-`Lucene <http://lucene.apache.org>`_ to index all string and timestamp
-information in the database, as well as all ``OriginalFiles`` which can
-be parsed to simple text (see :doc:`/developers/Search/FileParsers` for
+OMERO.server uses `Lucene <http://lucene.apache.org>`_ to index all string and
+timestamp information in the database, as well as all ``OriginalFiles`` which
+can be parsed to simple text (see :doc:`/developers/Search/FileParsers` for
 more information). The index is stored under :file:`/OMERO/FullText` (or the
 :file:`FullText` subdirectory of your :property:`omero.data.dir`, and can be
```
Comment (mtbc) [inline omero/developers/Modules/Search.txt:8]: just noticed we probably need a close-parenthesis before the comma

**Comment by hflynn** [comment]
@mtbc you happy for this to be merged or want someone else to look at it too?

File: omero/developers/Modules/Search.txt [hunk 3]
```diff
@@ -1,11 +1,10 @@
 OMERO search
 ============
 
-Beginning with 3.0-Beta3, the OMERO server will use
-`Lucene <http://lucene.apache.org>`_ to index all string and timestamp
-information in the database, as well as all ``OriginalFiles`` which can
-be parsed to simple text (see :doc:`/developers/Search/FileParsers` for
-more information). The index is stored under :file:`/OMERO/FullText` (or the
+OMERO.server uses `Lucene <http://lucene.apache.org>`_ to index all string and
+timestamp information in the database, as well as all ``OriginalFiles`` which
```
Comment (mtbc) [inline omero/developers/Modules/Search.txt]: maybe fix this `OriginalFile` too

**Comment by hflynn** [comment]
Done @mtbc - if there are no other issues, would you say this is good to merge now? Don't want to leave it hanging around for much longer really

**Comment by mtbc** [comment]
Good to merge.

**Comment by hflynn** [comment]
--rebased-to #1582

**Pull request merged**